src/data/list/duplicate.lean

```diff
@@ -0,0 +1,234 @@
+/-
+Copyright (c) 2021 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky, Chris Hughes
+-/
+import data.list.nodup
+import data.fin
+
+/-!
+# List duplicates
+
+## Main definitions
+
+* `list.duplicate x l : Prop` is an inductive property that holds when `x` is a duplicate in `l`
```

Comment:
make a note of the notation maybe?

Reply:
The notation is entirely local to the file.

Reply:
Added an implementation details note.